rust_src/src/fns.rs

```diff
@@ -32,3 +32,50 @@ fn featurep(feature: LispObject, subfeature: LispObject) -> LispObject {
         LispObject::constant_t()
     }
 }
+
+/// Announce that FEATURE is a feature of the current Emacs.
+/// The optional argument SUBFEATURES should be a list of symbols listing
+/// particular subfeatures supported in this version of FEATURE.
+#[lisp_fn(min = "1")]
+fn provide(feature: LispObject, subfeature: LispObject) -> LispObject {
+    feature.as_symbol_or_error();
+    subfeature.list_or_error();
+    unsafe {
+        if LispObject::from_raw(Vautoload_queue).is_not_nil() {
+            Vautoload_queue = Fcons(
+                Fcons(LispObject::from_fixnum(0).to_raw(), globals.f_Vfeatures),
+                Vautoload_queue,
+            )
+        }
+    }
+    let mut tem = memq(
```

Comment:
I'd be tempted to just inline this: `memq(...).is_nil()`. I know the C used `tem`, but I'm not sure it makes sense for us (it's not that helpful in the C either really).

Reply:
I think it makes sense as it's also used here

```
  tem = Fassq (feature, Vafter_load_alist);
  if (CONSP (tem))
    Fmapc (Qfuncall, XCDR (tem));
```